src/main/webapp/app/shared/components/atoms/upload-button/upload-button.component.html

```diff
@@ -1,9 +1,44 @@
-<button class="border border-neutral flex flex-col justify-center rounded-md p-2 w-full" [disabled]="true">
-  <p class="font-bold mt-1 mb-1">{{ uploadText() }}</p>
-  <div class="flex w-full justify-center">
-    <span class="flex border items-center justify-center rounded-full border-primary size-20">
-      <fa-icon [icon]="['fas', 'cloud-arrow-up']" class="background-primary" size="xl" />
-    </span>
-  </div>
-  <p class="mt-1 mb-1 text-sm">Click or drag and drop files to here to upload.</p>
+<button
+  class="border border-neutral flex flex-col justify-center rounded-md p-2 w-full h-full"
+  (click)="triggerFileInput()"
+  (dragover)="onDragOver($event)"
+  (dragleave)="onDragLeave($event)"
+  (drop)="onDrop($event)"
+  [class]="{ 'drag-over': isDragOver(), 'background-disabled': disabled() }"
+  [disabled]="disabled()"
+>
+  <input type="file" (change)="onFileSelected($event)" accept="application/pdf" multiple hidden #fileInput />
+  @if (disabled()) {
+    @if (documentIds(); as documentIds) {
+      <p class="font-bold mt-1 mb-1 text-color">
+        {{ documentIds.length }}
+        @if (documentIds.length === 1) {
+          <span jhiTranslate="entity.upload.uploaded_document"></span>
+        } @else if (documentIds.length > 1) {
+          <span jhiTranslate="entity.upload.uploaded_documents"></span>
+        } @else {}
+        <span [jhiTranslate]="`entity.upload.document_type.${ documentType() }`"></span>
+      </p>
+    }
+  } @else {
+    <p class="font-bold mt-1 mb-1 text-color" [jhiTranslate]="uploadKey()"></p>
+  }
+  @if (disabled()) {
+    <div class="flex w-full justify-center">
+      <span class="flex border items-center justify-center rounded-full border-success size-20">
+        <fa-icon [icon]="['fas', 'check']" class="background-success" size="xl" />
+      </span>
+    </div>
+  } @else {
+    <div class="flex w-full justify-center">
+      <span class="flex border items-center justify-center rounded-full border-primary size-20">
+        <fa-icon [icon]="['fas', 'cloud-arrow-up']" class="background-primary" size="xl" />
+      </span>
+    </div>
+    <p class="mt-1 text-sm text-color" jhiTranslate="entity.upload.instruction"></p>
+    <p class="mt-1 mb-1 text-xs text-color"><span jhiTranslate="entity.upload.max_size"></span> {{ maxUploadSizeInMb }}MB</p>
+  }
+  @if (isUploading()) {
+    <div><span jhiTranslate="entity.upload.uploaded_progress"></span>{{ uploadProgress() }}%</div>
```

Comment:
I am getting a translation error in the client for the uploaded progress text, which is caused by the misspelling here which doesn't match the key in the global.json file
```suggestion
    <div><span jhiTranslate="entity.upload.upload_progress"></span>{{ uploadProgress() }}%</div>
```